src/App.js

```diff
@@ -12,34 +12,55 @@ const App = () => {
     const [data, setData] = useState({})
     const [skills, setSkills] = useState([])
     const [selectedID, setSelectedID] = useState(1)
+    const [loading, setLoading] = useState(true)
+    const [error, setError] = useState(null)
+    const [searched, setSearched] = useState(false)
 
     useEffect(() => {
-        fetch("http://localhost:8080/jobs" + URL)
+        let url = "http://localhost:8080/jobs/recent"
+        if (searched) {
+            url = `http://localhost:8080/jobs?${searched}`
+        }
+        fetch(url)
             .then((res) => res.json())
             .then((res) => {
                 setDisplayJobs(res)
             })
-    }, [URL])
+    }, [searched])
 
     useEffect(() => {
+        setLoading(true)
         fetch("http://localhost:8080/jobs/" + selectedID)
-            .then (res => res.json())
+            .then(res => res.json())
             .then(data => {
                 setData(data)
                 setSkills(data.skills)
+                setLoading(false)
+            })
+            .catch(error => {
+                setLoading(false)
+                setError(error)
             })
     }, [selectedID])
 
 
     return (
         <>
             <Header/>
-            <DisplayJobs displayJobs={displayJobs} setDisplayJobs={setDisplayJobs}
-                         selectedID={selectedID} setSelectedID={setSelectedID}/>
-            <DetailModal data={data} setData={setData}
-                         skills={skills} setSkills={setSkills}
-                         selectedID={selectedID}/>
-            <Footer />
+
+            <SearchBar setURL={setURL}
+                       URL={URL}
+                       setSelectedID={setSelectedID}
+                       searched={searched}
+                       setSearched={setSearched}
+                       displayJobs={displayJobs}
+                       setDisplayJobs={setDisplayJobs}
+            />
+            <DetailModal data={data}
+                         skills={skills}
+                         loading={loading}
+                         error={error}/>
```

Comment:
Consistency: The component above has its `/>` on its own line, while this one doesn't.